### PR TITLE
{ACR} fix az acr build context match rules

### PR DESCRIPTION
**Pull request by chloeyin**
**Related command**
`az acr build`

**Description**

The current behavior of `az acr build` when passed a local context will parse the rule defined in the `.dockerignore` file and ignore the files match the pattern.  The parser doesn't honor the rule defined by `dockerignore` and will treat `**/` as `**`, which will incorrectly ignore some files that should not be excluded. For example: `**/bin` will be interpreted as `*bin` and match a folder like `vendor/google.golang.org/grpc/balancer/roundrobin`. The PR fixes this by adding slash when there is a `/` in the path.

**Testing Guide**

Unit tests with cases from how `dockerignore` file is parsed.

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
ACR

File: src/azure-cli/azure/cli/command_modules/acr/tests/latest/test_acr_archive_utils.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+import re
+import unittest
+
+from azure.cli.command_modules.acr._archive_utils import IgnoreRule
+
+class AcrArchiveUtilsTests(unittest.TestCase):
+    def test_ignore_rule(self):
+        test_data = (
+            ("**/bin", "user/roundrobin", False),
```
Comment (northtyphoon) [inline src/azure-cli/azure/cli/command_modules/acr/tests/latest/test_acr_archive_utils.py:14]: Can you add a test to verify 

**/bin, folder1/folder2/folder3/bin, True

Comment (chloeyin) [reply src/azure-cli/azure/cli/command_modules/acr/tests/latest/test_acr_archive_utils.py:15]: updated

**Review by northtyphoon — APPROVED** [review]
LGTM

**Comment by chloeyin** [comment]
@zhoxing-ms @jsntcy @yanzhudd 
Hi, can I get a review on this PR? Thanks

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**